File: Android/tuilivekit/src/main/java/com/trtc/uikit/livekit/liveroom/data/LiveRoomInfo.java
package com.trtc.uikit.livekit.liveroom.data;

import static com.trtc.uikit.livekit.liveroom.core.TUILiveDefine.InteractionType.BROADCAST;

import androidx.annotation.Nullable;

import com.trtc.tuikit.common.livedata.LiveData;
import com.trtc.uikit.livekit.common.core.LiveDefine;
import com.trtc.uikit.livekit.common.utils.Constants;
import com.trtc.uikit.livekit.liveroom.core.TUILiveDefine.InteractionType;
import com.trtc.uikit.livekit.liveroom.core.TUILiveDefine.UserLiveStatus;

import java.util.Objects;
import java.util.concurrent.CopyOnWriteArraySet;

public class LiveRoomInfo {
    public String roomId;

    public RoomConfig getRoomConfig() {
        return roomConfig;
    }

    public LiveData<String>                             name                = new LiveData<>("");
    public LiveData<String>                             coverURL            =
            new LiveData<>(Constants.DEFAULT_COVER_URL);
    public LiveData<String>                             category            = new LiveData<>("");
    public LiveData<LiveDefine.LiveStreamPrivacyStatus> liveMode            =
            new LiveData<>(LiveDefine.LiveStreamPrivacyStatus.PUBLIC);
    public LiveData<InteractionType>                    interactionType     = new LiveData<>(BROADCAST);
    public UserInfo                                     anchorInfo          = new UserInfo();
    public LiveData<CopyOnWriteArraySet<UserInfo>>      audienceList        =
            new LiveData<>(new CopyOnWriteArraySet<>());
    public LiveData<Integer>                            audienceCount       = new LiveData<>(0);
    public LiveData<CopyOnWriteArraySet<UserInfo>>      linkingAudienceList =
            new LiveData<>(new CopyOnWriteArraySet<>());
    public LiveData<UserLiveStatus>                     userLiveStatus      = new LiveData<>(UserLiveStatus.NONE);
    public RoomConfig                                   roomConfig          = new RoomConfig();
    public CopyOnWriteArraySet<String>                  hasVideoList        = new CopyOnWriteArraySet<>();
    public CopyOnWriteArraySet<String>                  hasAudioList        = new CopyOnWriteArraySet<>();

    public LiveRoomInfo(String roomId) {
        this.roomId = roomId;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj instanceof LiveRoomInfo) {
            return this.roomId.equals(((LiveRoomInfo) obj).roomId);
        }
        return false;
    }

    @Override
    public int hashCode() {
        return Objects.hash(roomId);
    }
}

File: iOS/TUILiveKit/Source/LiveRoom/Data/LiveRoomInfo.swift
//
//  LiveRoomInfo.swift
//  TUILiveKit
//
//  Created by WesleyLei on 2023/12/7.
//

import Foundation

class LiveRoomInfo {
    let roomId: Observable<String> = Observable("")
    let name: Observable<String> = Observable("")
    let coverUrl: Observable<String> = Observable("")
    let category: Observable<LiveStreamCategory> = Observable(.chat)
    let liveMode: Observable<LiveMode> = Observable(.public)
    let interactionType: Observable<InteractionType> = Observable(.broadcast)
    let anchorInfo: Observable<UserInfo> = Observable(UserInfo())
    let audienceList: Observable<[UserInfo]> = Observable([])
    let audienceCount: Observable<Int> = Observable(0)
    let linkingAudienceList: Observable<[UserInfo]> = Observable([])
    let roomConfig: RoomConfig = RoomConfig()
    let userLiveStatus: Observable<UserLiveStatus> = Observable(.none)
    var maxSeatCount:Int = 0
    var giftIncome:Int = 0
    var giftPeopleMap:[String:String] = [:]
    var createTime:TimeInterval = 0
    lazy var userInfoPool : UserInfoPool = {
        return UserInfoPool(roomId: roomId.value)
    }()
    
    init(roomId: String) {
        self.roomId.value = roomId
    }
    
    init() {
    }
}
